rubicon/objc/api.py

```diff
@@ -851,17 +828,31 @@ def __setattr__(self, name, value):
                 if isinstance(value, enum.Enum):
                     value = value.value
                 ObjCBoundMethod(method, self)(value)
-            else:
-                # Add a new Python attribute.
-                # These attributes are currently only stored on the Python ObjCInstance object
-                # and not on the actual Objective-C object.
-                # To ensure that the attributes aren't lost when the ObjCInstance has no more Python references
-                # (but the Objective-C object still has Objective-C references
-                # and may be wrapped in an ObjCInstance again in the future),
-                # every ObjCInstance with additional Python attributes is "kept alive" on the Python side.
-                # This prevents the ObjCInstance from being destroyed while the Objective-C object still exists.
-                self._ensure_keep_alive()
+            elif name.startswith("_"):
                 super(ObjCInstance, type(self)).__setattr__(self, name, value)
+            else:
+
+                # Wrap the Python object as an WrappedPyObject instance.
+                # A reference will be retained as long as the WrappedPyObject is alive.
+                wrapper = WrappedPyObject.alloc().initWithObjectId_(id(value))
```

Comment:
This is more legible for the time being but should probably be replaced by the appropriate `send_message` calls.

Reply:
I've replaced this with the appropriate `send_message` calls.